rest/data/perf_test.go

```diff
@@ -15,6 +16,63 @@ import (
 
 const testDBName = "cedar_rest_data_test"
 
+type MockQueue struct {
```

Comment:
I think you can refactor your code such that the components produce the input (the output of the query), produce jobs, and enqueue them, and avoid the need for this mock. 

You can use the local queue, (and maybe with a noop-flag) on the job to ensure that the jobs enqueue properly, though I wouldn't. The goal of your testing should be with the behavior of the code, and I think it's safe to trust that your dependencies (e.g. the queue) behaves as you'd expect.

Reply:
don't I need some kind of mocked queue in order to assert that my code did in fact enqueue the job I expected it to?

Reply:
you can use a local queue, and ensure that it's added. (either by checking the number of jobs before and after you call the function,) or by checking the results of the queue. 

In general, though, you can trust that the Put method works.

Reply:
I mean, I do trust that it works, that's why I've mocked the thing! Is there some reason not to mock things in golang?

Reply:
I don't think it provides benefit and I'd prefer to not mock things out except in some lower level cases. 

Please remove it.

Reply:
What can I mock at a lower level such that I don't need a queue locally and can detect that this job gets scheduled as I expect it to?

Reply:
I'd just use https://godoc.org/github.com/mongodb/amboy/queue#NewLocalLimitedSize

and just check the size of the queue before and after.

Reply:
I've done this, but I think it would be really nice to have some way of checking the TimeSeriesIds of the jobs that have been enqueued.